src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -81,6 +82,26 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Sorts a shadow list of persons by name in ascending order.
+     * Note that the original list is not modified.
+     *
+     * @return an unmodifiable List view with elements sorted and cast as immutable
+     */
+    public List<ReadOnlyPerson> sort() {
+        List<Person> shadowList = internalList;
+        shadowList.sort(new Comparator<Person>() {
```

Comment:
How about implementing such sorting in OOP fashion? This method is more like FP.